packages/components/src/mobile/image/index.native.js

```diff
@@ -0,0 +1,224 @@
+/**
+ * External dependencies
+ */
+import { Image, Text, TouchableWithoutFeedback, View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Icon } from '@wordpress/components';
+import { image as icon } from '@wordpress/icons';
+import { usePreferredColorSchemeStyle } from '@wordpress/compose';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { MediaEdit } from '../media-edit';
+import { getImageWithFocalPointStyles } from '../image-with-focalpoint';
+import styles from './style.scss';
+import SvgIconRetry from './icon-retry';
+import SvgIconCustomize from './icon-customize';
+
+const ICON_TYPE = {
+	PLACEHOLDER: 'placeholder',
+	RETRY: 'retry',
+	UPLOAD: 'upload',
+};
+
+function editImageComponent( { open, mediaOptions } ) {
+	return (
+		<TouchableWithoutFeedback onPress={ open }>
+			<View style={ styles.editContainer }>
+				<View style={ styles.edit }>
+					{ mediaOptions() }
+					<Icon
+						size={ 16 }
+						icon={ SvgIconCustomize }
+						{ ...styles.iconCustomise }
+					/>
+				</View>
+			</View>
+		</TouchableWithoutFeedback>
+	);
+}
+
+const ImageComponent = ( {
+	align,
+	alt,
+	isSelected,
+	isUploadFailed,
+	isUploadInProgress,
+	onSelectMediaUploadOption,
+	openMediaOptions,
+	retryMessage,
+	url,
+	width: imageWidth,
+	height: imageHeight,
+	focalPoint,
+} ) => {
+	const [ imageData, setImageData ] = useState( null );
+	const [ containerSize, setContainerSize ] = useState( null );
+
+	useEffect( () => {
+		if ( url ) {
+			Image.getSize( url, ( imgWidth, imgHeight ) => {
+				setImageData( {
+					aspectRatio: imgWidth / imgHeight,
+					width: imgWidth,
+					height: imgHeight,
+				} );
+			} );
+		}
+	}, [ url ] );
+
+	const onContainerLayout = ( event ) => {
+		const { height, width } = event.nativeEvent.layout;
+		setContainerSize( { width, height } );
```

Comment:
Maybe it's worth to check if heigh and width are different than current? Thanks to that we avoid unnecessary re-renders :)

Reply:
Definitely! We don't want to render things when we don't want to, I added that check. Thanks!